From bdd195792b7752e4c5a0ff74cb5b01a6accee517 Mon Sep 17 00:00:00 2001
From: Jan Paul Batrina <jpmbatrina01@gmail.com>
Date: Thu, 25 Mar 2021 19:31:39 +0800
Subject: [PATCH] Use QPalette::highlight for the scrollbar minimap slider

Previously, QPalette::link() was used which is the color for hyperlinks.
Most themes set this to a shade of blue, so the slider will almost
always be blue even when, for example, the main color
of a theme is green.

The selection highlight color was chosen because it is also the color
used for indicating the active tab and is, generally, often attuned
to a theme's main color.

BUG: 434690
---
 src/view/kateviewhelpers.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/view/kateviewhelpers.cpp b/src/view/kateviewhelpers.cpp
index 365e07e0..48e94609 100644
--- a/src/view/kateviewhelpers.cpp
+++ b/src/view/kateviewhelpers.cpp
@@ -750,7 +750,7 @@ void KateScrollBar::miniMapPaintEvent(QPaintEvent *e)
     // calculate colors
     const QColor backgroundColor = m_view->defaultStyleAttribute(KTextEditor::dsNormal)->background().color();
     const QColor foregroundColor = m_view->defaultStyleAttribute(KTextEditor::dsNormal)->foreground().color();
-    const QColor highlightColor = palette().link().color();
+    const QColor highlightColor = palette().highlight().color();
 
     const int backgroundLightness = backgroundColor.lightness();
     const int foregroundLightness = foregroundColor.lightness();
-- 
GitLab

